### ISSUE TITLE
test_interrupt_kills_child failing in 0.11.0rc3

### ISSUE THREAD

**Issue by katre** [bazelbuild/bazel#4625]
https://ci.bazel.build/blue/organizations/jenkins/Global%2Fbazel-tests/detail/bazel-tests/500/tests

Test is failing on the platforms:
- linux-x86_64,python_version=2
- linux-x86_64,python_version=3
- ubuntu_16.04-x86_64

Error:
```
INFO: $TEST_TMPDIR defined: output root default is '/home/ci/.cache/bazel/_bazel_ci/65ed5a6a0442d902ec235fee508231f8/bazel-sandbox/1483943166230936815/execroot/io_bazel/_tmp/2e8cdad9647bde18379f08ad523677f3' and max_idle_secs default is '15'.
INFO: $TEST_TMPDIR defined: output root default is '/home/ci/.cache/bazel/_bazel_ci/65ed5a6a0442d902ec235fee508231f8/bazel-sandbox/1483943166230936815/execroot/io_bazel/_tmp/2e8cdad9647bde18379f08ad523677f3' and max_idle_secs default is '15'.
Loading:
Loading: 0 packages loaded
Analyzing: target //foo:sleep-minute (10 packages loaded)
INFO: Analysed target //foo:sleep-minute (10 packages loaded).
INFO: Found 1 target...
[0 / 4] [-----] Creating source manifest for //foo:sleep-minute
Target //foo:sleep-minute up-to-date:
bazel-bin/foo/sleep-minute
INFO: Elapsed time: 1.451s, Critical Path: 0.08s
INFO: Build completed successfully, 4 total actions
INFO: Running command line: bazel-bin/foo/sleep-minute
-- Test log: -----------------------------------------------------------
------------------------------------------------------------------------
test_interrupt_kills_child FAILED: Shell script still running after SIGINT sent to server .
/home/ci/.cache/bazel/_bazel_ci/65ed5a6a0442d902ec235fee508231f8/bazel-sandbox/1483943166230936815/execroot/io_bazel/bazel-out/k8-fastbuild/bin/src/test/shell/integration/run_test.runfiles/io_bazel/src/test/shell/integration/run_test:206: in call to test_interrupt_kills_child
INFO[run_test 2018-02-12 16:26:54 (+0000)] Cleaning up workspace
```

**Comment by katre** [comment]
Blocking release https://github.com/bazelbuild/bazel/issues/3959

**Comment by katre** [comment]
Failure is repeatable, but the entire test must be run with the RC, as so:
```
$ bazel build //src:bazel
...
INFO: Build completed successfully, 1 total action

$ cp bazel-bin/src/bazel /tmp/bazel-0.11.0rc3

$ /tmp/bazel-0.11.0rc3 test //src/test/shell/integration:run_test
....
INFO: Build completed, 1 test FAILED, 1244 total actions
//src/test/shell/integration:run_test                                    FAILED in 1 out of 3 in 76.5s
  Stats over 3 runs: max = 76.5s, min = 17.0s, avg = 36.9s, dev = 28.0s
    ERROR   .test_interrupt_kills_child (61.5s)

Executed 1 out of 1 test: 1 fails locally.
```

**Comment by katre** [comment]
Test passes with the above procedure if I revert 28bd997c1c8793973f63dcae4c22bbae49e7d8b7.

**Comment by katre** [comment]
Linking to #4608.

**Comment by agoulti** [comment]
Thanks for the repro, looking into this

**Comment by agoulti** [comment]
I've rolled back the original change to get this green again. The problem is more subtle than I initially thought, so I don't have an ETA of when I'll get to the bottom of it.


If I understand this correctly, test-setup.sh does not affect anything in the "bazel run" behavior.
So, the object of the test ("SIGINT kills the underlying command in bazel run") shouldn't have been affected.

We are only modifying the wrapper under which the test itself was ran, and the way the test was ran. I'm still trying to figure out what effect it might have.

An obvious thought is, we added a level of indirection (the test is now ran as a subprocess of test-setup.sh rather than inside that process), which might make a difference; however, in my attempts, simply adding such a level of indirection wasn't enough to fail the test.

I will continue with the investigation tomorrow.
